### DISCUSSION TITLE
Prop request: Form field with connected fields

### DISCUSSION THREAD

**Discussion by kylegilbreath** [twilio-labs/paste#1195] [Ideas] [answered]
👋 heyo my fav Paste team! I've run into a few occasions when translating a legacy Consoles pattern (i.e. Status Callback URL input with GET/POST select) into Paste components where an input is paired with another select and needs to be visually represented in the UI as being connected somehow. Using separate labels on each form element seems to visually disconnect them from each other, so I've been suggesting we add an additional select next to the Input without a visible Label (we've been discussing this [here](https://github.com/twilio-labs/paste/discussions/1188) for more context). But aside from this specific example, I'm sure we'll run into other scenarios similar to this where we'll have to represent related (or connected) form fields (inputs/selects) and it might be helpful to have a specific component or modification on an existing Input or Select that would allow us to attach an addition field to it. That way we could have a consistent visual styling on how the connected fields should look across Console (i.e. consistent spacing in between the connected fields) and also have the Label be represented in the DOM correctly as the label for both fields. Right now we are kind of hacking this by just not showing a label on one of the form elements, but I'm sure this isn't super great for screen readers.

<img width="663" alt="Screen Shot 2021-02-12 at 3 59 39 PM" src="https://user-images.githubusercontent.com/12417517/108106400-20cb8900-704b-11eb-9968-48d1de9bd3ce.png">

<img width="879" alt="Screen Shot 2021-02-03 at 12 43 39 PM" src="https://user-images.githubusercontent.com/12417517/108106475-38a30d00-704b-11eb-8ada-84a67a9a4312.png">

Anyway, I was exploring how other design systems handle this type of scenario, and it looks like Polaris has a _Text field with connected field_ example of their [Text Field](https://polaris.shopify.com/components/forms/text-field) that shows the addition of a `connectedLeft` prop to allow this this type of pattern to be added to a Text Field. I'm not sure if this has been brought up before, but thought I would throw the idea out there in case it hasn't. Anyway, I would love to add more context if needed. Thanks!

<img width="885" alt="Screen Shot 2021-02-16 at 11 23 29 AM" src="https://user-images.githubusercontent.com/12417517/108107997-3e015700-704d-11eb-81fb-3c9a03d80ffa.png">

**Comment by richbachman** [comment]
Hi @kylegilbreath thanks for the suggestion. I'll share with the team and get back with you.

**Comment by richbachman** [comment, marked as answer]
@kylegilbreath the team talked about this today. It looks like Polaris isn't meeting the accessibility guidelines on that example. Each input must be labelled. Their example kind of works, but its a bit of a stretch in that it is implicit that weight has a unit of measure. I think a lot of the accessibility concerns have been discussed in previous discussions: https://github.com/twilio-labs/paste/discussions/1188 and https://github.com/twilio-labs/paste/discussions/993.

We'll either add guidance to the docs or develop a pattern  to help make these types of connected fields more consistent. If you'd like to contribute to that guidance or pattern, let us know. Thanks!
